src/overlays/actors/ovl_Obj_Kendo_Kanban/z_obj_kendo_kanban.c

```diff
@@ -217,238 +233,251 @@ void ObjKendoKanban_Destroy(Actor* thisx, PlayState* play) {
 }
 
-void func_80B65D68(ObjKendoKanban* this, PlayState* play) {
+void ObjKendoKanban_Settled(ObjKendoKanban* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
 
-    if (func_80B6618C(this, play) == 1) {
-        player->unk_AC0 = 700.0f;
+    if (ObjKendoKanban_IsPlayerOnTop(this, play) == true) {
+        player->displacementY = 700.0f;
     }
 }
 
-void func_80B65DA8(ObjKendoKanban* this, PlayState* play) {
-    Vec3f sp5C;
+void ObjKendoKanban_HandlePhysics(ObjKendoKanban* this, PlayState* play) {
+    Vec3f rootCornerPos = this->cornerPos[0];
     s32 pad[2];
-    Vec3f sp48;
-    s32 pad2;
-    s32 index = 0;
+    Vec3f vecCenterOut = this->actor.world.pos;
+    const s16 MAX_BOUNCE_COUNT = 7;
+    s32 idxRootCornerPos = 0;
     s32 i;
-    f32 sp38;
-
-    sp5C = this->unk_26C[0];
-    sp48 = this->actor.world.pos;
-
-    sp48.x -= this->unk_2D8.x;
-    sp48.y -= this->unk_2D8.y;
-    sp48.z -= this->unk_2D8.z;
-
-    sp38 = (this->unk_2F0.x * sp48.z) + (this->unk_2F0.z * -sp48.x);
-
-    if (sp38 < 0.0f) {
-        this->unk_304 += 0x64;
+    f32 verticalScalar;
+
+    // Calculate an affect on the rotation from gravity.
+    vecCenterOut.x -= this->centerPos.x;
+    vecCenterOut.y -= this->centerPos.y;
+    vecCenterOut.z -= this->centerPos.z;
+    verticalScalar = (this->rotationalAxis.x * vecCenterOut.z) + (this->rotationalAxis.z * -vecCenterOut.x);
+    if (verticalScalar < 0.0f) {
+        this->rotationVelocity += 0x64;
     } else {
-        this->unk_304 -= 0x64;
+        this->rotationVelocity -= 0x64;
     }
 
-    for (i = 0; i < ARRAY_COUNT(this->unk_26C); i++) {
-        if (this->unk_26C[i].y < sp5C.y) {
-            sp5C = this->unk_26C[i];
-            index = i;
+    // Find the lowest point
+    for (i = 0; i < ARRAY_COUNT(this->cornerPos); i++) {
+        if (this->cornerPos[i].y < rootCornerPos.y) {
+            rootCornerPos = this->cornerPos[i];
+            idxRootCornerPos = i;
         }
     }
 
-    if (index != this->unk_2FC) {
-        this->unk_300 = 1;
-        this->unk_2FC = index;
-        this->unk_2E4 = this->unk_29C[index];
+    // When the lowest point changes, re-initialize the actor position to that point
+    if (idxRootCornerPos != this->idxLastRootCornerPos) {
+        this->bHasNewRootCornerPos = true;
+        this->idxLastRootCornerPos = idxRootCornerPos;
+        this->rootCornerPos = this->cornerPoints[idxRootCornerPos];
 
         Matrix_Push();
         Matrix_SetTranslateRotateYXZ(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z,
                                      &this->actor.shape.rot);
         Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
-        Matrix_MultVec3f(&this->unk_2E4, &this->actor.world.pos);
-        this->actor.world.pos = sp5C;
+        Matrix_MultVec3f(&this->rootCornerPos, &this->actor.world.pos);
+        this->actor.world.pos = rootCornerPos;
```

Comment:
Would like another set of eyes on this. Unless there's a side-effect of the Matrix_* operations I'm not understanding, it would appear that actor.world.pos could simply be set to rootCornerPos and the entire bundle of Matrix_* calls could be eliminated?

Reply:
I know at least `Matrix_Scale` with modify the current matrix on the stack (however it also gets popped of later, so I'm not sure). @EllipticEllipsis would probably know more.

Reply:
This is very strange-looking. It creates a new scale-rotate-translate that sets up the transformation for the current actor coordinates. It applies it to the lowest corner's position, writes that to world.pos. And then overwrites that with the `rootCornerPos` almost immediately, and drops the matrix. The whole function looks quite redundant already, too.

So yes, afaics this is a waste of time.

Reply:
One thing you could do is remove the matrix calls, and see if the actor behaves differently.